include/experimental/fundamental/v3/product_type/ap.hpp

```diff
@@ -112,6 +112,21 @@ namespace product_type
           product_type::element_sequence_for<ProductType>{});
   }
 
+  //! struct mapping a product_type to an Applicative.
```

Comment:
I believe this is not correct as for pair and tuple.

This is why I defined n-functors and n-applicatives.